xmtp_mls/src/identity/xmtp_id/mod.rs

```diff
@@ -0,0 +1,114 @@
+pub mod identity;
+
+use crate::storage::identity_inbox::StoredIdentity;
+use crate::{api::ApiClientWrapper, builder::ClientBuilderError, storage::EncryptedMessageStore};
+use crate::{xmtp_openmls_provider::XmtpOpenMlsProvider, Fetch};
+pub use identity::Identity;
+use log::debug;
+use log::info;
+use xmtp_cryptography::signature::sanitize_evm_addresses;
+use xmtp_id::InboxId;
+use xmtp_proto::api_client::{XmtpIdentityClient, XmtpMlsClient};
+
+/// The member that the [Identity] is created from.
+pub enum Member {
+    /// user's Ethereum wallet address
+    Address(String),
+    /// address and corresponding legacy private key(same secp256k1 as Ethereum wallet)
+    LegacyKey(String, Vec<u8>),
+}
+
+pub enum IdentityStrategy {
+    /// Tries to get an identity from the disk store, if not found creates an identity, will create an inbox if address isn't associated with one already.
+    ///
+    /// Platform SDK could have following logic before `IdentityStrategy.initialize_identity(api_client, store)`:
+    ///
+    /// inbox_id = libxmtp.get_inbox_id(wallet_address)
+    /// if !inbox_id {
+    ///     inbox_id = libxmtp.generate_inbox_id(wallet_address, 0)
+    ///     if use_legacy_key() {
+    ///         pass // We will create inbox & add association in one call in `IdentityStrategy.initialize_identity(api_client, store)`.
+    ///     } else {
+    ///         inbox_id = libxmtp.generate_inbox_id(wallet_address, random_nonce)
+    ///         libxmtp.create_new_inbox_with_user_signature(inbox_id, wallet_address)
+    ///         TODO: Similar to use_legacy_key(), we can include this in one call so that user doesn't have to sign twice.
```

Comment:
@neekolas what do you think about this TODO?

Reply:
Definitely fine to leave as a TODO. We can discuss in our sync tomorrow.